### PR TITLE
Move strict_tls_security header from server level to location level

### PR DESCRIPTION
**Pull request by lei-tang**
Move strict_tls_security header from server level to location level.

### CODE REVIEW

File: start_esp/nginx-auto.conf.template [hunk 1]
```diff
@@ -194,6 +189,11 @@ http {
     # End CORS settings
 % endif
 
+% if enable_strict_transport_security:
+      # Enable HSTS (HTTP Strict Transport Security), 31536000 is 1 year.
+      add_header Strict-Transport-Security "max-age=31536000; includeSubdomains;";
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply start_esp/nginx-auto.conf.template]: Added "always" so tls is always required.

File: start_esp/test/testdata/expected_enable_strict_transport_security_nginx.conf [hunk 2]
```diff
@@ -86,6 +83,9 @@ http {
 
 
 
+      # Enable HSTS (HTTP Strict Transport Security), 31536000 is 1 year.
+      add_header Strict-Transport-Security "max-age=31536000; includeSubdomains;" always;
+
```
Comment (guimarca) [inline start_esp/test/testdata/expected_enable_strict_transport_security_nginx.conf]: This does not solve the issue, nginx only applies the `add_header` directives in the 'deepest' level of indentation, ignoring all in the previous ones. Read [this](https://github.com/cloudendpoints/esp/issues/473#issuecomment-436558755).

Comment (lei-tang) [reply start_esp/test/testdata/expected_enable_strict_transport_security_nginx.conf]: @guimarca The PR has been updated to place the add_header directives to other places. Can you check again?

**Comment by lei-tang** [comment]
@qiwzhang Presubmit tests passed. Can you approve again?

**Comment by guimarca** [comment]
Do you have an estimate for when will this be added to the official release?

**Pull request merged**